tasks/10_checks-repos.yml

```diff
@@ -0,0 +1,120 @@
+
+- name: Group quota on alternc www directory mount
+  pause:
+    prompt: >
+        The partition mounted where will be created the html directory for AlternC (by default 
+        /var/www/alternc) has no group quota management. You won't be able to manage AlternC users' quotas. 
+        Do you want to continue?
+  when: not ansible_facts.mounts | json_query( alternc_html_mount_query ) | first | regex_search('(grpjquota=aquota\.group|grpquota)')
+
+
+- name: When we want to set a favicon.ico, check if icoutils is installed
+  block: 
+
+  - name: Gather the list of local packages
+    local_action: package_facts
+    become: no
+    register: local_packages
+  
+  - debug: 
+      msg: '{{ local_packages | json_query("ansible_facts.packages.icoutils") }}'
+      verbosity: 2
+
+  - name: If icoutils not installed
+    block: 
+
+    - name: Stop if icoutils is not installed
+      pause:
+        prompt: |
+          You want to set a custom favicon.ico and icoutils is not installed on your local controller. 
+          Do you want to install it? 
+          Abort if no, and run the present playbook with -K (--ask-become-pass) option, to provide your sudo password.
+
+## I couldn't sort out to ask the sudo password only when the local package is absent: 
+#    - name: Ask for "ansible_become_password"
+#      pause:
+#        prompt: "Please enter your password for sudo"
+#        echo: no
+#      register: ansible_become_pass_input
+
+    - name: Install icoutils in local controller
+      apt:
+        name: icoutils
+        state: present
+      delegate_to: localhost
+
+    when: local_packages | json_query("ansible_facts.packages.icoutils") | type_debug != "list"
+
+  when: alternc_favicon is string and alternc_favicon != '' 
+
+- debug: 
+    msg: "{{ ansible_distribution }}-{{ ansible_distribution_release }}.yml"
+    verbosity: 2
+
+- name: Include variables dependind on debian version 
+  include_vars: "{{ ansible_distribution }}-{{ ansible_distribution_release }}.yml"
+
+- name: Adding AlternC repository key
+  apt_key:
+    url: "{{ alternc_repo }}/{{ alternc_key }}"
+    state: present
+
+- name: Adding AlternC repository
+  apt_repository: 
+    repo: "deb {{ alternc_repo }} {{ alternc_branch }}"
+    filename: alternc
+    state: present
+  register: alternc_repo_add
+
+- block: 
+  - name: Define stable as preferred debian distribution
+    lineinfile: 
+      path: /etc/apt/apt.conf.d/30StableRelease
+      line: 'APT::Default-Release "stable";'
+      create: yes
+      state: present
+
+  - name: If distribution is buster, add Sid repository for phpmyadmin
+    apt_repository: 
+      repo: "deb {{ alternc_debian_sid_repo }} {{ alternc_debian_sid_branch }}"
+      state: present
+    register: debian_repo_sid_add
+  when: ( ansible_distribution == 'Debian') and (ansible_distribution_release == 'buster')
+  
+- name: Update packages indexes
+  apt:
+    update_cache: yes
+    cache_valid_time: 0
+  when: > 
```

Comment:
The idea is to run the update_cache if at least one new repo is added. The package install later will perform their own update if needed. 
But: seeing that [the default value of of `update_cache` is 0](https://docs.ansible.com/ansible/2.9/modules/apt_module.html#parameter-cache_valid_time), we should add to all apt install tasks something like: 
```
cache_valid_time: 3600
```
to avoid repeatigng cache update.

Reply:
But if packages list were recently updated (on repo server), the next installation task will fail. [These task](https://github.com/UdelaRInterior/ansible-AlternC/blob/36b431e4c0b8e1cfe35bb71903353d8df0c23d1e/tasks/20_debconf.yml#L5), for example.

Reply:
As [default value of `cache_valid_time`](https://docs.ansible.com/ansible/2.9/modules/apt_module.html#parameter-cache_valid_time) is 0, [the task you point out](https://github.com/UdelaRInterior/ansible-AlternC/blob/36b431e4c0b8e1cfe35bb71903353d8df0c23d1e/tasks/20_debconf.yml#L5) *will* perform an apt update. 

In fact, in the logic I want to implement, I missed it: it should have a `cache_valid_time: 3600` as [this task](https://github.com/UdelaRInterior/ansible-AlternC/blob/36b431e4c0b8e1cfe35bb71903353d8df0c23d1e/tasks/30_packages.yml#L10) and following ones. (In the new organisation of task files, it shoud be in [the same file](https://github.com/UdelaRInterior/ansible-AlternC/blob/36b431e4c0b8e1cfe35bb71903353d8df0c23d1e/tasks/30_packages.yml) - in the previous order of tasks for packages install, there were problem for sid dependencies).

Again: the idea is that the role performs one, and only one, cache update, if a new repo was added or if the cache is more than 1h old (and we could put any arbitrary delay, just longer than the maximum execution time of the role). 

I'm not a repo's expert so I'm not 100% sure of what follows, but I think that if the repo is update in the less than an hour between the cache update and the package installation, this task will fail only if the version we pretend to install has disappeared, what should not happen for the `latest` version of a well maintained package. If one of our packages is concerned by the repo's update, we will then just installed the previous and not the latest version. And the new version of the package will be installed when we re-run the role, either because we have been informed about or routinely.

Reply:
It may happen that there is a package that has been removed from the repo. So, when you run an `apt install`, you will have a `404 Error'. To prevent that, you should run `apt update` before.

Reply:
If a package that we use is removed form the repo, our role is broken. So running an `apt update` is then pointless. I understand that you say update the cache each time apt module is ran, so what's the meaning of `cache_valid_time` parameter?

To attend your concern I propose we reduce the delay of `cache_valid_time` from 1 hour to a few minutes, but I consider it's unuseful to run several times in the role's execution an update of the cache. Even if you are right, the probability is very very low. And when a playbook fails, we stop and analyse why. Here, re-running the playbook should be enough (except if our set of packages is broken in the distro, but then it will happen anyway).

Reply:
What i mean: if the version that is in the local cache of apt is updated and removed from the repo, when you run `apt install` you get the 404 error. And `apt` will ask you: did you run `apt update`? 
The cache_valid_time option will be required in every task with the apt module. But it would only be necessary to run `apt update` before any task with the apt module.
It's just a detail....

Reply:
I propose my changes [here](https://github.com/UdelaRInterior/ansible-AlternC/pull/11)

Reply:
Sorry, I omitted to commit some `cache_valid_time: 3600` Do you agree that  f617850 reasonably performs `apt update` to avoid any blocking situation?